server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseBackgroundSyncAction.java

```diff
@@ -113,9 +114,19 @@ public void backgroundSync(
                     ActionListener.wrap(
                             r -> {},
                             e -> {
-                                if (ExceptionsHelper.unwrap(e, AlreadyClosedException.class, IndexShardClosedException.class) == null) {
-                                    getLogger().warn(new ParameterizedMessage("{} retention lease background sync failed", shardId), e);
+                                final TransportException maybeTransport =
+                                        (TransportException) ExceptionsHelper.unwrap(e, TransportException.class);
+                                if (maybeTransport != null
+                                        && (maybeTransport.getMessage().equals("TransportService is closed stopped can't send request")
+                                        || maybeTransport.getMessage().equals("transport stopped, action: " + ACTION_NAME + "[p]"))) {
```

Comment:
this shares some code with `ReplicationOperation.onNoLongerPrimary`. As long as we don't have a dedicated exception, might be good to share this in a helper method.